apps/prairielearn/src/question-servers/freeform.js

```diff
@@ -1260,6 +1348,15 @@ module.exports = {
       const courseIssues = [];
       const context = await module.exports.getContext(question, course);
 
+      // Hack: we need to propagate this back up to the original caller so
+      // they can expose the selected renderer to the client via a header, but
+      // parent functions don't actually return things. So we'll just stick it
+      // in the `locals` object that the parent will be able to read from.
+      //
+      // See the `setRendererHeader` function in `lib/question` for where this
+      // is actually used.
+      locals.question_renderer = context.renderer;
```

Comment:
🤮 but eventually we'll rewrite all this code to explicitly propagate things both down and up the call stack.